packages/realm-react/src/__tests__/UserProvider.test.tsx

```diff
@@ -0,0 +1,114 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+import React, { useEffect, useState } from "react";
+import { AppProvider } from "../AppProvider";
+import { renderHook, waitFor } from "@testing-library/react-native";
+
+import { AppConfigBuilder } from "@realm/app-importer";
+import { useEmailPasswordAuth } from "../useEmailPasswordAuth";
+import { baseUrl, importApp } from "./helpers";
+import { AuthOperationName } from "../types";
+import { UserProvider, useUser } from "../UserProvider";
+import { App } from "realm";
+
+const testEmail = "test@email.com";
+const testPassword = "password";
+
+let userRegistered = false;
```

Comment:
Even though the variable is updated to `true` later on, it doesn't look like the variable is (currently) needed (only passed as the initial value of `isUserRegistered`).

Reply:
I wanted to commit a comment, but had moved to other things.  This ensures that subsequent tests do not have to register.  Only the first that renders this will.